crates/turborepo-lib/src/task_graph/visitor.rs

```diff
@@ -177,18 +190,119 @@ impl<'a> Visitor<'a> {
             // hashing so that downstream tasks can count on the hash existing
             //
             // bail if the script doesn't exist
-            let Some(command) = command else { continue };
+            let Some(_command) = command else { continue };
 
             let output_client = self.output_client();
+            let continue_on_error = self.opts.run_opts.continue_on_error;
             let prefix = self.prefix(&info);
             let pretty_prefix = self.color_cache.prefix_with_color(&task_hash, &prefix);
             let ui = self.ui;
+            let manager = self.manager.clone();
+            let package_manager = self.package_graph.package_manager().clone();
+            let workspace_directory = self.repo_root.resolve(workspace_dir);
+            let errors = errors.clone();
+            let task_id_for_display = self.display_task_id(&info);
 
             tasks.push(tokio::spawn(async move {
+                let task_id = info;
                 let _task_cache = task_cache;
                 let mut prefixed_ui =
-                    Self::prefixed_ui(ui, is_github_actions, &output_client, pretty_prefix);
-                prefixed_ui.output(command);
+                    Self::prefixed_ui(ui, is_github_actions, &output_client, pretty_prefix.clone());
+
+                let mut cmd = Command::new(package_manager.to_string());
+                cmd.args(["run", task_id.task()]);
```

Comment:
Note that in this PR, properly setting up the environment is not handled. Opting to do this later down the line in it's own PR to make this one more contained.